.github/workflows/unit-tests.yml

```
@@ -0,0 +1,15 @@
+name: Unit-Tests
+
+on:
+  pull_request:
+    branches: [ main ]
+
```

```suggestion
  push:
    branch: [main]
```
Could we add a trigger for pushes in main. So if we don't have a pull request but fixing something in main the test will run too.